front/lib/front/models/commit_job.ex

```diff
@@ -146,18 +178,65 @@ defmodule Front.Models.CommitJob do
     |> Enum.filter(fn elem -> elem != :skip end)
   end
 
+  defp generate_git_commands(params) do
+    [
+      # Configure the branch that checkout should use
+      "export SEMAPHORE_GIT_BRANCH=#{params.initial_branch}",
+      # Clone repository using the read-only deploy key
+      "checkout",
+      # switch to a new branch if user configured a different target branch
+      configure_git_target_branch(params.initial_branch, params.target_branch),
+      add_commands_for_modified_files(params),
+      add_commands_for_deleted_files(params),
+      # Configure the user to be author of the changes
+      "git config --global user.name #{params.user.name}",
+      "git config --global user.email #{params.user.email}",
+      # Create commit with changes
+      "git add .",
+      "git commit -m \"#{params.commit_message}\"",
+      # Push commit to remote repository
+      "git push $SEMAPHORE_GIT_URL #{params.target_branch} --force",
+      # save the new commit sha in artifacts
+      "git rev-parse HEAD > commit_sha.val",
+      "artifact push job commit_sha.val -d .workflow_editor/commit_sha.val"
+    ]
+    |> List.flatten()
+    |> Enum.filter(fn elem -> elem != :skip end)
+  end
+
+  defp configure_git_target_branch(initial, target) do
+    if initial == target do
+      :skip
+    else
+      [
+        # Fetch remote branches
+        "git fetch origin #{target} || true",
+        # Check if branch exists on remote
+        "if git show-ref --verify --quiet refs/remotes/origin/#{target}; then",
+        "  git checkout -B #{target} origin/#{target}",
+        "else",
+        "  git checkout -B #{target} #{initial}",
+        "fi"
```

Comment:
I'm not 100%sure how this multi-line if-else will work on Zebra, but I assume it would be interpreted as multiple one-line commands. Is this working as expected when tested manually?

Reply:
I think I can do a one-liner from this